lib/rom/setup/auto_registration_strategies/no_namespace.rb

```diff
@@ -1,13 +1,14 @@
 require 'pathname'
 
 require 'dry/core/inflector'
+require 'rom/types'
 require 'rom/setup/auto_registration_strategies/base'
 
 module ROM
   module AutoRegistrationStrategies
     class NoNamespace < Base
-      option :directory, reader: true, type: Pathname
-      option :entity, reader: true, type: Symbol
+      option :directory, reader: true, type: PathnameType
```

Comment:
@solnic just for the clearance: you could use a "simplified" syntax as well (I've added it some time ago for compatibility to reform etc.)

```ruby
option :directory, PathnameType, reader: true
```